enterprise/frontend/src/embedding-sdk/types/plugins.ts

```diff
@@ -0,0 +1,52 @@
+import type { ReactNode } from "react";
+
+import type { DashCardMenuItem } from "./dashboard";
+import type { Question } from "./question";
+
+export type MetabaseClickAction = {
+  name: string;
+} & Record<string, any>;
+
+export type MetabaseDataPointObject = {
+  value?: string | number | null | boolean;
+  column?: Record<string, any>;
+  event?: MouseEvent;
+  data?: {
+    col: Record<string, any> | null;
+    value: string | number | null | boolean;
+  }[];
+};
+
+export type MetabaseClickActionPluginsConfig = (
+  clickActions: MetabaseClickAction[],
+  clickedDataPoint: MetabaseDataPointObject,
+) => MetabaseClickAction[];
+
+export type DashboardCardMenuCustomElement = ({
+  question,
+}: {
+  question: Question;
+}) => ReactNode;
+
+export type CustomDashboardCardMenuItem = ({
+  question,
+}: {
+  question?: Question;
+}) => DashCardMenuItem;
+
+export type DashboardCardCustomMenuItem = {
+  withDownloads?: boolean;
+  withEditLink?: boolean;
+  customItems?: (DashCardMenuItem | CustomDashboardCardMenuItem)[];
+};
+
+export type MetabaseDashboardPluginsConfig = {
```

Comment:
Should we have the `Sdk` prefix across these types: `SdkDashboardPluginsConfig` ?

Reply:
My proposal is when we deal with just placement of types - rename some of them in a consistent way

Reply:
Agreed with re-naming, let's get it to one prefix 👍🏻